### ISSUE TITLE
Empty item in env field inside c_cpp_properties.json causes error

### ISSUE THREAD

**Issue by schperplata** [microsoft/vscode-cpptools#2532]
**Type: LanguageService**
Issue Type: <b>Bug</b>

I'm playing around with using VS Code for embedded project. I have created a c_cpp_properties.json file, where I am planing to list all directories and files inside of different "env" fields and than passing this values to "configuration" "includePath", "browse" and "define" fields.

The problem specifically is caused by an empty field, followed by a non empty field, see example below:
```
{
    "env": {
        ...
        "userDefines": [],
        ...
        "cubeMXDefines": [
            "USE_HAL_DRIVER",
            "STM32F078xx"
        ],
        ...
    },
    "configurations": [
        {
            ...
            "defines": [
                "${userDefines}",
                "${cubeMXDefines}"
            ],
            ...
        }
    ],
    "version": 4
}
```
As you can see, **userDefines** field is empty, while **cubeMXDefines** in not. While editing main.c for example, I get an error in problems tab:
![defines error](https://user-images.githubusercontent.com/8736111/45594826-5fa01500-b9a1-11e8-92ac-38e4c9818bcd.PNG)

If I write anything into **userDefines** field, Intellisense works as expected. 
Is there something I am missing, or is this a bug (not handled empty env field)?

VS Code version: Code 1.27.2 (f46c4c469d6e6d8c46f268d1553c5dc4b475840f, 2018-09-12T16:17:45.060Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-2520M CPU @ 2.50GHz (4 x 2494)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.95GB (3.16GB free)|
|Process Argv|C:\Users\domen\AppData\Local\Programs\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|1.0.60
CppSnippets|har|0.0.13
cortex-debug|mar|0.1.21
python|ms-|2018.8.0
cpptools|ms-|0.18.1
debug|web|0.22.0

</details>

**Comment by bobbrow** [comment]
The replacement logic is using a "falsy" check to see if a replacement should be made.  It probably needs to handle an empty array separately.

**Comment by sean-mcmanus** [comment]
This empty array environment variable bug is fixed in 0.19.0.
